### PR TITLE
Update Moshi version

### PR DESCRIPTION
**Pull request by MickaelGuilbeaud**
# Description

This PR does multiple things:
- Update Gradle and tools dependencies so we can use the latest Gradle APIs.
- Update Retrofit and Moshi versions.
- Fix a bug with a nullable field marked as non-nullable.
- Fix a Lint error in AspectRatioImageView.

# Tests

Tested in the example demo application and in our Medium app that the changes works as expected.

### CODE REVIEW

File: photopicker/src/main/java/com/unsplash/pickerandroid/photopicker/data/UnsplashUser.kt [hunk 1]
```diff
@@ -13,7 +13,7 @@ data class UnsplashUser(
     val location: String?,
     val total_likes: Int,
     val total_photos: Int,
-    val total_collection: Int,
+    val total_collection: Int?,
```
Comment (grosalex) [inline photopicker/src/main/java/com/unsplash/pickerandroid/photopicker/data/UnsplashUser.kt:16]: why is this becoming nullable?

**Review by grosalex — APPROVED** [review]
Let's fork it to move faster but on long term what about opening a pr on the main repo ?

File: photopicker/src/main/java/com/unsplash/pickerandroid/photopicker/data/UnsplashUser.kt (hunk 1 above)
Comment (MickaelGuilbeaud) [reply photopicker/src/main/java/com/unsplash/pickerandroid/photopicker/data/UnsplashUser.kt:16]: Because the Unsplash API can return null for this field.

**Comment by MickaelGuilbeaud** [comment]
> Let's fork it to move faster but on long term what about opening a pr on the main repo ?

@grosalex There is an 1 year old PR for that with no response from the maintainers: https://github.com/unsplash/unsplash-photopicker-android/pull/16, so the library is no longer maintained.

File: photopicker/build.gradle [hunk 2]
```diff
@@ -43,7 +48,7 @@ dependencies {
     // misc
     implementation 'com.squareup.picasso:picasso:2.71828'
```
Comment (prcaen) [inline photopicker/build.gradle:49]: :(, so we have Glide, Coil, Picasso for loading images.

**Review by prcaen — APPROVED** [review]

File: photopicker/build.gradle (hunk 2 above)
Comment (MickaelGuilbeaud) [reply photopicker/build.gradle:49]: Sadly, but we can update Unsplash to use Coil in the future to remove the Picasso depency. My guess is that is should be straightforward.

**Pull request merged**